zebra/zebra_nhg.c

```diff
@@ -1047,25 +1048,24 @@ int zebra_nhg_kernel_del(uint32_t id)
 }
 
 /* Some dependency helper functions */
-static struct nhg_hash_entry *depends_find(struct nexthop *nh, afi_t afi)
+static struct nhg_hash_entry *depends_find(const struct nexthop *nh, afi_t afi)
 {
-	struct nexthop *lookup = NULL;
+	struct nexthop lookup;
 	struct nhg_hash_entry *nhe = NULL;
 
 	if (!nh)
 		goto done;
 
-	copy_nexthops(&lookup, nh, NULL);
-
-	/* Clear it, in case its a group */
-	nexthops_free(lookup->next);
-	nexthops_free(lookup->prev);
-	lookup->next = NULL;
-	lookup->prev = NULL;
+	/* Capture a snapshot of this single nh; it might be part of a list,
+	 * so we need to make a standalone copy.
+	 */
+	memset(&lookup, 0, sizeof(lookup));
+	nexthop_copy(&lookup, nh, NULL);
```

Comment:
no need to set `next` and `prev` to `NULL`, I take it?

Reply:
yes - I've memset() it, so those pointers should stay NULL

Reply:
it doesnt need to set it to NULL anymore because he is calling the newer `nexthop_copy()` api.

it doesnt copy the list pointers, just he single nexthop.